opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/log/LoggingAutoConfiguration.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.log;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.autoconfig.TracerAutoConfiguration;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Pavol Loffay
+ */
+@Configuration
+@ConditionalOnBean(Tracer.class)
+@AutoConfigureAfter(TracerAutoConfiguration.class)
+@ConditionalOnProperty(name = "opentracing.spring.cloud.log.enabled", havingValue = "true", matchIfMissing = true)
+public class LoggingAutoConfiguration {
+
+  @Bean
```

Comment:
I presume that you are using a `@Bean` not because you explicitly need to inject the `SpanLogsAppender`, but simply because it's convenient point to add the appender to root logger. Is that correct?

Reply:
Correct, we don't need a bean here. Is there a better way to do it?

Reply:
I am looking into how it can be done without a bean. I'll let you know soon enough :)

Reply:
You can replace the `@Bean` method with the following code:

```java
  @Autowired
  private Tracer tracer;

  @PostConstruct
  public void setupAppender() {
      SpanLogsAppender spanLogsAppender = new SpanLogsAppender(tracer);
      Logger rootLogger = getRootLogger();
      rootLogger.addAppender(spanLogsAppender);
  }
```
The effect is exactly the same as far as logging and avoid adding the Appender to the Spring Application Context

Reply:
@geoand thanks. I will use `PostConstruct`. I was wondering if it's possible to return void from bean method. but I didn't find anything on this.